packages/app-store/components.tsx

```diff
@@ -13,15 +13,27 @@ import type { RouterOutputs } from "@calcom/trpc/react";
 import type { App } from "@calcom/types/App";
 import { Icon } from "@calcom/ui";
 
+//import setDefaultConferencingApp from "./_utils/setDefaultConferencingApp";
 import { InstallAppButtonMap } from "./apps.browser.generated";
 import type { InstallAppButtonProps } from "./types";
 
 export const InstallAppButtonWithoutPlanCheck = (
   props: {
     type: App["type"];
+    defaultInstall?: boolean;
+    slug?: string;
   } & InstallAppButtonProps
 ) => {
-  const mutation = useAddAppMutation(null);
+  const mutation = useAddAppMutation(null, {
+    onSuccess: () => {
+      if (me && props.defaultInstall && props.slug) {
+        setDefaultConferencingApp.mutate({ slug: props.slug });
+      }
```

Comment:
This change makes InstallAppButton do too many things. Following the Open/Closed principle, I think it would be much better to introduce an `onSuccess` prop to the component and then let the onboarding component provide it's own onSuccess callback that would take care of the default installation.

Reply:
We can then remove `defaultInstall` prop and `slug` prop